google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractStructReader.java

```diff
@@ -251,7 +251,6 @@ public Date getDate(String columnName) {
 
   @Override
   public Value getValue(int columnIndex) {
-    checkNonNull(columnIndex, columnIndex);
```

Comment:
Is this a breaking change? We are modifying the contract here, I wonder if any customer is relying on this? I would imagine not, but double checking.

Reply:
Hmm... That's a good point. I don't think anyone would be relying on it, but you're right that it is a change of the contract. In this case we will always be returning a non-null value, even if the database value is `NULL` (we're returning a `Value` instance whose internal value is null and whose type is set to the type of the column), so it also a safe value to return. I think there are two (three) options here:
1. Accept the small change in contract.
2. Add an extra method named something like 'getValueOrNull`
3. Do a major version bump (but I don't feel like this feature is worth that)

@thiagotnunes @rajatbhatta @ansh0l Any specific thoughts on this?